google/cloud/pubsub_v1/opentelemetry_tracing.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020, Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from contextlib import contextmanager
+
+from google.api_core.exceptions import GoogleAPICallError
+
+_LOGGER = logging.getLogger(__name__)
+
+try:
+    from opentelemetry import trace
+    from opentelemetry import propagators
+    from opentelemetry.trace import SpanContext
+    from opentelemetry.trace import get_current_span
+    from opentelemetry.trace import set_span_in_context
+    from opentelemetry.trace.status import Status
+    from opentelemetry.instrumentation.utils import http_status_to_canonical_code
+
+    USE_OPENTELEMETRY = True
+except ImportError:
+    _LOGGER.info(
+        "This service supports OpenTelemetry, but OpenTelemetry could"
+        "not be imported. To use OpenTelemetry, please install the"
+        "opentelemetry-api, opentelemetry-sdk, and opentelemetry-instrumentation"
+        "pip modules. See also"
+        "https://opentelemetry-python.readthedocs.io/en/stable/getting-started.html"
+    )
+    USE_OPENTELEMETRY = False
+    pass
+
+
+@contextmanager
+def create_span(span_name, attributes=None, parent=None):
+    """ Creates a new span
+
+    Args:
+        span_name (str): the name of the new span
+        attributes ([dict], optional): A dictionary
```

Comment:
(nit)
Following the `mypy` notation, optional types are specified as `Optional[typename]`

Applies to multiple places.